### DISCUSSION TITLE
KeyError: 'annotation_type_list' in pthr_go_annots.py when gene list includes specific genes

### DISCUSSION THREAD

**Discussion by ariadnamorales** [geneontology/helpdesk#371] [Q&A]
Hello,

I am running **pthr_go_annots.py** in "geneinfo" mode and I noticed that when specific genes (e.g. ANKRD40CL) are part of the input gene lists, the run crashes with the following error:

Traceback (most recent call last):
  File "pthr_go_annots.py", line 227, in <module>
    response.print_results()
  File "pthr_go_annots.py", line 126, in print_results
    for dt in self.handle_annotation_data_type(r['annotation_type_list']['annotation_data_type']):
KeyError: 'annotation_type_list'

When I remove such gene from the input list, there is no error. Any ideas why this is happening? The gene has a valid panther annotation:
http://pantherdb.org/tools/gxIdsList.do?list=upload_1&organism=Homo%20sapiens

I have identified other gene symbols with similar cases.

Best,

Ariadna

**Comment by suzialeksander** [comment]
Hi @ariadnamorales =, 

Apologies for the delay, we were on winter break.  Have you read the PANTHER documentation or contacted PANTHER about this? http://pantherdb.org/help/PANTHERhelp.jsp

**Comment by kltm** [comment]
@ariadnamorales 
PANTHER also has an issue helpdesk at https://github.com/pantherdb/Helpdesk/issues. As this is possibly a software issue, it might be good to try this issue over there.

**Comment by dustine32** [comment]
Hi @ariadnamorales!

I'll definitely take a look at what's going on. Since this code is maintained in [pantherdb/pantherapi-pyclient](https://github.com/pantherdb/pantherapi-pyclient) a good place for the issue is https://github.com/pantherdb/pantherapi-pyclient/issues.

**Comment by dustine32** [comment]
@ariadnamorales I opened an issue [here](https://github.com/pantherdb/pantherapi-pyclient/issues/6) and pushed a quick fix. Let me know if you still get the error after a `git pull`.
